api/src/modules/scenarios/cost-surface/cost-surface.facade.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@nestjs/common';
+import { Request, Response } from 'express';
+import { AdjustCostSurface } from '../../analysis/entry-points/adjust-cost-surface';
+import { ProxyService } from '../../proxy/proxy.service';
+
+@Injectable()
+export class CostSurfaceFacade {
+  constructor(
+    private readonly adjustCostSurfaceService: AdjustCostSurface,
+    private readonly proxyService: ProxyService,
+  ) {}
+
+  /**
+   * non blocking - will do job in "background"
+   *
+   * @param scenarioId
+   * @param request
+   */
+  convert(scenarioId: string, request: Request) {
+    // TODO #0 Generate & Dispatch Api Event (some wrapping service for /dummy/"terminating" if already running)
+
+    // TODO #1 Call Proxy Service to get GeoJSON
+    // this.proxyService.... - modify this to send back data, not act on Response
+
+    // TODO #2 Call Analysis-module with scenario id & GeoJson from above
+    this.adjustCostSurfaceService
+      .update(scenarioId, {
+        geo: {
+          features: [],
+          type: 'FeatureCollection',
+        },
```

Comment:
this could probably be a plain list of ids and associated cost value

Reply:
I would refrain from putting more logic there. We already have `analysis module` which handles validation of integrity and persistence - the facade is "only" connecting those two pieces together (i.e. rest->shapefile->geo->process)

Reply:
Revisited in https://github.com/Vizzuality/marxan-cloud/pull/174/commits/1a12fbbe7d4bdf4df095a88d5dd69ced127a860c